v1/bundle/store.go

```diff
@@ -1149,3 +1186,35 @@ func ActivateLegacy(opts *ActivateOpts) error {
 	opts.legacy = true
 	return activateBundles(opts)
 }
+
+// GetActivator returns the Activator registered under the given id
+func GetActivator(id string) (Activator, error) {
+	activator, ok := activators[id]
+
+	if !ok {
+		return nil, fmt.Errorf("no activator exists under id %s", id)
+	}
+
+	return activator, nil
+}
+
+// RegisterActivator registers a bundle Activator under the given id.
+// The id value can later be referenced in ActivateOpts.Plugin to specify
+// which activator should be used for that bundle activation operation.
+// Note: This must be called *before* RegisterDefaultBundleActivator.
+func RegisterActivator(id string, a Activator) {
+	activatorMtx.Lock()
+	defer activatorMtx.Unlock()
+
+	if id == defaultActivatorID {
+		panic("cannot use reserved activator id, use a different id")
+	}
+
+	activators[id] = a
+}
+
+func init() {
+	activators = map[string]Activator{
+		defaultActivatorID: &DefaultActivator{},
+	}
+}
```

Comment:
Is that not basically the same as 
```suggestion
var activators = map[string]Activator{
		defaultActivatorID: &DefaultActivator{},
	}
```

Reply:
I did a little bit of digging and found some interesting bits near the end of the Go Programming Language Spec's [Package Initialization section](https://go.dev/ref/spec#Package_initialization) (with relevant bits in **bold**):
> The entire package is initialized by **assigning initial values to all its package-level variables** followed by **calling all init functions** in the order they appear in the source, possibly in multiple files, as presented to the compiler.

So, it seems the second version you've suggested is actually better overall-- the variable will show up initialized in every usage site, and gets initialized before any `init` functions run. I'll make the change shortly!